app/channel.go

```diff
@@ -2599,6 +2599,28 @@ func (a *App) IsCRTEnabledForUser(c request.CTX, userID string) bool {
 	return threadsEnabled
 }
 
+func (a *App) ValidateUserPermissionsOnChannels(c request.CTX, session model.Session, e model.EmailInvite) {
+	var allowedChannelIds []string
+	channels := e.GetChannels()
+
+	for _, channelId := range channels {
+		channel, err := a.GetChannel(c, channelId)
+		if err != nil {
+			mlog.Info("Invite users to team - couldn't get channel " + channelId)
+			continue
+		}
+
+		if channel.Type == model.ChannelTypePrivate && a.SessionHasPermissionToChannel(c, session, channelId, model.PermissionManagePrivateChannelMembers) {
```

Comment:
Are we able to instead use `HasPermissionToChannel` instead of `SessionHasPermissionToChannel` so that we don't have to create the dummy session in `CreateUserWithToken`?

Reply:
sure!